examples/spytial_annotations_demo.rs

```diff
@@ -0,0 +1,108 @@
+use rust_viz::spytial_annotations::{
+    SpytialDecoratorsBuilder, AnnotationBuilder, HasSpytialDecorators,
+    annotate_instance, collect_decorators_for_instance, to_yaml_for_type, to_yaml_for_instance,
+    SpytialDecorators,
+};
+
+// Example struct with manual decorator implementation
+#[derive(Debug)]
+struct Node {
```

Comment:
Not quite how I'd want it.

In Python:

```
@attribute(field="id")
class Node:
    id: int                    #
    lo: Optional["Node"]       # 
    hi: Optional["Node"]       #

```

And in Rust:
```
#[...whatever the rust equivalent...]
class Node:
    id: int                    # stable integer id (for debugging / refs)
    v: Optional[str]         # None for constants; otherwise variable name
    lo: Optional["Node"]       # 0-edge (None for constants)
    hi: Optional["Node"]       # 1-edge (None for constants)

    def is_const(self) -> bool:
        return self.v is None
```